extensions/amazon-aws/src/glue-jobs.tsx

```diff
@@ -0,0 +1,56 @@
+import { GlueClient, ListJobsCommand } from "@aws-sdk/client-glue";
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { isReadyToFetch, resourceToConsoleLink } from "./util";
+import { AwsAction } from "./components/common/action";
+
+export default function GlueJobs() {
+  const { data: glueJobs, error, isLoading, revalidate } = useCachedPromise(fetchJobs);
+
+  return (
+    <List
+      isLoading={isLoading}
+      searchBarPlaceholder="Filter Jobs by name..."
+      searchBarAccessory={<AWSProfileDropdown onProfileSelected={revalidate} />}
+    >
+      {error ? (
+        <List.EmptyView title={error.name} description={error.message} icon={Icon.Warning} />
+      ) : (
+        glueJobs?.map((job) => <GlueJob key={job} job={job} />)
+      )}
+    </List>
+  );
+}
+
+function GlueJob({ job: glueJobName }: { job: string }) {
+  return (
+    <List.Item
+      key={glueJobName}
+      icon={"aws-icons/glue.png"}
+      title={glueJobName || ""}
+      actions={
+        <ActionPanel>
+          <AwsAction.Console url={resourceToConsoleLink(glueJobName, "AWS::Glue::JobRuns")} />
+          <ActionPanel.Section title={"Copy"}>
+            <Action.CopyToClipboard title="Copy Job Name" content={glueJobName || ""} />
+          </ActionPanel.Section>
+        </ActionPanel>
+      }
+      accessories={[{ text: glueJobName || "" }]}
```

Comment:
suggestion: this accessory is unnecessary

Reply:
Also, in an effort to bridge the cap with the console, usually with the commands we have been trying to show as much metadata for the resource item as possible. I know [ListJobs](https://docs.aws.amazon.com/glue/latest/webapi/API_ListJobs.html) just give you the name of the jobs, but you can then [GetJob](https://docs.aws.amazon.com/glue/latest/webapi/API_GetJob.html) to map some key elements as details/metadata. The setup will look very similar to DynamoDB command, if you want to use that as a reference.

Reply:
I agree, we can choose not to add scope creep to this PR. But that's my one caveat with the current state of AWS commands - it is just an entry point to direct me to console, and then I have to do everything on the console. However, if the profile has permissions, why can't we perform operations directly on raycast.

I think you are in a better position to think and design overall UX and actions that you frequently use with Glue jobs and try to add it to raycast. I can guide you through the process but I am not using Glue jobs much now-a-days. We can definitely follow it up in a separate issue/PR discussion.

Reply:
But we should at least try to structure the API calls as hooks and handle pagination through raycast native protocol to ensure we remain within memory and throttle constraints.